src/store/validation/index.ts

```diff
@@ -0,0 +1,105 @@
+  flowValidator: ValidateFunction;
+}
+
+export const stateFactory = (): IValidationState => ({
+  blockValidationStatuses: {} as { [key:string]: IValidationStatus },
+  blockTypeValidators: {} as { [key: string]: ValidateFunction },
+  flowValidationStatus: {} as IValidationStatus,
+  flowValidator: {} as ValidateFunction
+})
+
+export const getters: GetterTree<IValidationState, IRootState> = {
+
+}
+
+export const mutations: MutationTree<IValidationState> = {
+
+}
+
+export const actions: ActionTree<IValidationState, IRootState> = {
+  async validate_block({ state }, { block } : { block: IBlock }): Promise<IValidationStatus> {
+    const { uuid: blockId, type: blockType } = block
+    const blockTypeWithoutNameSpace = blockType.split('.')[blockType.split('.').length - 1]
+    const blockJsonSchemaFile = `I${blockTypeWithoutNameSpace}Block.json`
+
+    if (isEmpty(state.blockTypeValidators) || !state.blockTypeValidators.hasOwnProperty(blockTypeWithoutNameSpace)) {
+      // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+      state.blockTypeValidators[blockTypeWithoutNameSpace] = createDefaultJsonSchemaValidatorFactoryFor(require(`../../../_tmp/${blockJsonSchemaFile}`))
+    }
+    const validate = state.blockTypeValidators[blockTypeWithoutNameSpace]
+
+    state.blockValidationStatuses[blockId] = {
+      isValid: validate(block),
+      errors: validate.errors,
+    }
+
+    return state.blockValidationStatuses[blockId]
+  },
+
+  async validate_flow({ state }, { flow } : { flow: IFlow }): Promise<IValidationStatus> {
+    if (isEmpty(state.flowValidator)) {
+      // TODO: point to the right JSON once we consume the right flow-runner version, then delete tmp file
+      state.flowValidator = createDefaultJsonSchemaValidatorFactoryFor(require('../../../_tmp/flowSpecJsonSchema.json'))
+    }
+    const validate = state.flowValidator
+
+    state.flowValidationStatus = {
+      isValid: validate(flow),
+      errors: validate.errors,
+    }
+
+    return state.flowValidationStatus
+  }
+}
+
+export const store: Module<IValidationState, IRootState> = {
+  namespaced: true,
+  state: stateFactory,
+  getters,
+  mutations,
+  actions,
+}
+
+export default store
+
+/**
+ * Create AJV Validator
+ * Usage :
+ * const validate = createDefaultJsonSchemaValidatorFactoryFor(require('./some-json-schema.json')
+ * const isValid = validate(myData)
+ * const error = validate.errors
+ *
+ * @param jsonSchema
+ */
+export function createDefaultJsonSchemaValidatorFactoryFor(jsonSchema: JSONSchema7): ValidateFunction {
+  return ajv.compile(jsonSchema)
+}
+
+export function debutValidationStatus(status: IValidationStatus, customMessage: string) {
+  console.debug(
+    customMessage,
+    ' | isValid:', status.isValid,
+    ' | errors:', `${status.hasOwnProperty('errors') && !!status.errors! ? (status.errors!).map(item => get(item, 'message', 'undefined')).join(';') : 'undefined'}`,
```

Comment:
Another proposal, I'd suggest we implement i18n in Beta version https://ajv.js.org/api.html#errors-i18n

If you're ok @bzabos , I can create a ticket for that

Reply:
I think this is a great idea! Thanks @safydy !

Reply:
Great ! created the ticket https://viamoinc.atlassian.net/browse/VMO-3957